pkg/querier/querier.go

```diff
@@ -119,58 +121,71 @@ func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Qu
 }
 
 // NewQueryable creates a new Queryable for cortex.
-func NewQueryable(dq, cq storage.Queryable, distributor Distributor, cfg Config) storage.Queryable {
+func NewQueryable(distribQuerier, storeQuerier storage.Queryable, chunkIterFn chunkIteratorFunc, cfg Config) storage.Queryable {
 	return storage.QueryableFunc(func(ctx context.Context, mint, maxt int64) (storage.Querier, error) {
 		q := querier{
-			distributor: distributor,
 			ctx:         ctx,
 			mint:        mint,
 			maxt:        maxt,
+			chunkIterFn: chunkIterFn,
+		}
+
+		dqr, err := distribQuerier.Querier(ctx, mint, maxt)
+		if err != nil {
+			return nil, err
 		}
 
+		q.distributorQuerier = dqr
+
 		// Include ingester only if maxt is within queryIngestersWithin w.r.t. current time.
 		now := model.Now()
 		if cfg.QueryIngestersWithin == 0 || maxt >= int64(now.Add(-cfg.QueryIngestersWithin)) {
-			dqr, err := dq.Querier(ctx, mint, maxt)
-			if err != nil {
-				return nil, err
-			}
-			q.queriers = append(q.queriers, dqr)
+			q.selectQueriers = append(q.selectQueriers, dqr)
 		}
 
 		// Include store only if mint is within QueryStoreAfter w.r.t current time.
 		if cfg.QueryStoreAfter == 0 || mint <= int64(now.Add(-cfg.QueryStoreAfter)) {
-			cqr, err := cq.Querier(ctx, mint, maxt)
+			cqr, err := storeQuerier.Querier(ctx, mint, maxt)
 			if err != nil {
 				return nil, err
 			}
 
-			q.queriers = append(q.queriers, cqr)
+			q.selectQueriers = append(q.selectQueriers, cqr)
 		}
 
 		return q, nil
 	})
 }
 
 type querier struct {
-	queriers []storage.Querier
+	// used for labels and metadata queries
+	distributorQuerier storage.Querier
 
-	distributor Distributor
+	// used for selecting series
+	selectQueriers []storage.Querier
```

Comment:
Can we do better than "selectQueriers" ?  What is the difference between a select and a query, and how does that change when we do both together?

Reply:
There is only `Select` operation on `Querier` type. I wanted to express that *only* these queriers are used in `Select`, unlike `metadataQuerier`. But I think with `metadataQuerier`, I will simply rename this back.